MetalLB make-ready: edits, structure, formatting, etc. (#165)
Mirantis/mke-docs: content/docs/configuration/metallb/_index.md

- A MetalLB-compatible cluster network configuration.

- `kube-proxy` running in `iptables` mode.

- The absence of any cloud provider configuration.

## Configuration

You configure MetalLB through the `Addons` section of the MKE 4 configuration
file. The function is disabled by default, and thus to use MetalLB you must set
the `enabled` parameter to `true`.

1. Obtain the default configuration file.

   ```bash
   mkectl init
   ```

   The example that follows presents the MetalLB configuration section as it is used to deploy MetalLB as a Helm chart within the cluster:

   ```yaml
   spec:
     addons:
       - chart:
           name: metallb
           repo: https://metallb.github.io/metallb
           values: |
             controller:
               tolerations:
                 - key: node-role.kubernetes.io/master
                   operator: Exists
                   effect: NoSchedule
             speaker:
               frr:
                 enabled: false
           version: 0.14.7
         dryRun: false
         enabled: false
         kind: chart
         name: metallb
         namespace: metallb-system
   ```

> 

Reviewer: Add a callout: FRR mode is disabled by default.

---
title: MetalLB load balancer
weight: 3
---

MetalLB is a load balancer designed for bare metal Kubernetes clusters that uses standard routing protocols.

## Prerequisites

- An MKE cluster that does not already have load-balancing functionality.

- A MetalLB-compatible cluster network configuration.

> - `kube-proxy` running in `iptables` mode.

Reviewer 2: ```suggestion
- A `kube-proxy` component running in `iptables` mode.
```

Listed as component in the official doc: https://kubernetes.io/docs/concepts/overview/components/#node-components

Author: Yes, kube-proxy is a Kubernetes component, but drawing it out as such reads _awkward_. I would go with 'instance' before 'component'. That said, though, it does bring up a question. @sakshisharma84, should be saying "A kube-proxy instance running in `iptables` mode on each cluster node."?

- The absence of any cloud provider configuration.

## Configuration

> You configure MetalLB through the `Addons` section of the MKE 4 configuration

Reviewer 2: ```suggestion
You can configure MetalLB through the `Addons` section of the MKE 4 configuration
```

file. The function is disabled by default, and thus to use MetalLB you must set
the `enabled` parameter to `true`.

> 1. Obtain the default configuration file.

Reviewer 2: ```suggestion
1. Obtain the default configuration file:
```

   ```bash
   mkectl init
   ```

>    The example that follows presents the MetalLB configuration section as it is used to deploy MetalLB as a Helm chart within the cluster:

Reviewer 2: ```suggestion
   The following example presents the MetalLB configuration section used to deploy MetalLB as a Helm chart within the cluster:
```

Author: The edit changes the emphasis of the sentence.

| enabled                             | Enablement state of add-on.                             | false                                                                                     |
| name                                | Name of the add-on.                                     | metallb                                                                                   |
| namespace                           | Namespace used for<br>deploying MetalLB.                   | metallb-system                                                                            |
| chart.name                          | Name of the MetalLB Helm chart.                         | metallb                                                                                   |
| chart.repo                          | MetalLB Helm repository.                                | https://metallb.github.io/metallb                                                         |
| chart.version                       | Version of the MetalLB Helm chart.                      | 0.14.7                                                                                    |

> | chart.values.<br>controller.tolerations | Tolerations for the<br>MetalLB controller pod.             | YAML: <br><br> - key: node-role.kubernetes.io/master<br>   &nbsp; operator: Exists <br>  &nbsp;   effect: NoSchedule |

Reviewer 3: Unify into a single string, and add backticks: `chart.values.controller.tolerations`

> | chart.values.<br>speaker.frr.enabled    | Enablement state of FRR<br>with regard to MetalLB speaker. | false                                                                                     |

Reviewer 3: Unify into a single string and add backticks: `chart.values.speaker.frr.enabled`

and add backticks: `false`

An MKE version comparison for MetalLB configuration parameters is offered in the following table:

| MKE-3                                                          | MKE-4                                   |
|----------------------------------------------------------------|-----------------------------------------|
| `[cluster_config.metallb_config.enabled]`                        | `addons.metallb.enabled`                  |
| `[[cluster_config.metallb_config.metallb_ip_addr_pool]]`         | `Deprecated`                            |

## IP address pool configuration

>   <callout type="info"> Support for IP address pool configuration is deprecated in MKE 4, and thus you must now independently manage the configuration. </callout>

Reviewer 2: ```suggestion
  <callout type="info"> Support for IP address pool configuration is deprecated in MKE 4, and thus you must now manage the configuration independently. </callout>
```

Check if the note is wrapped in the output, if not, divide it between two lines.

### Create IP address pools in a fresh installation

>   For information on how to create IP address pools, refer to the official MetalLB documentation, [Layer 2 configuration](https://metallb.universe.tf/configuration/#layer-2-configuration).

Reviewer 2: ```suggestion
  For information on how to create IP address pools, refer to the official MetalLB documentation,
  [Layer 2 configuration](https://metallb.universe.tf/configuration/#layer-2-configuration).
```

spec:
  addresses:
  - 192.168.10.0/24
---
apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
  name: empty
  namespace: metallb-system
```

As necessary, refer to the official MetalLB documentation, [Advanced AddressPool](https://metallb.universe.tf/configuration/_advanced_ipaddresspool_configuration).

### Managing IP address pools during upgrade

 During an upgrade from MKE 3 to MKE 4, if metalLB is enabled in the former the
 configured IP address pool details display in the upgrade summary.

The following example configuration presents MetalLB enabled in MKE 3:

```toml
[cluster_config.metallb_config]
  enabled = true

  [[cluster_config.metallb_config.metallb_ip_addr_pool]]
    name = "example1"
    external_ip = ["192.168.10.0/24", "192.168.1.0/24"]
  [[cluster_config.metallb_config.metallb_ip_addr_pool]]
    name = "example2"
    external_ip = ["192.155.10.0/24"]
```

The upgrade summary presents as follows:

```bash
MetalLB IP address pools
---------------
name: example1
ipaddress: [192.168.10.0/24 192.168.1.0/24]

name: example2
ipaddress: [192.155.10.0/24]

Please make sure that you create these pools after MKE4 installation is complete.
```

> Refer to the upgrade summary for guidance in the creation of IP address pools. To create the pools, you can use the template that follows or consult the official MetalLB documentation [Layer 2 configuration](https://metallb.universe.tf/configuration/#layer-2-configuration) for assistance.

Reviewer 2: ```suggestion
Refer to the upgrade summary for guidance in the creation of IP address pools. To create the pools,
you can use the template that follows or consult the official MetalLB documentation
[Layer 2 configuration](https://metallb.universe.tf/configuration/#layer-2-configuration) for assistance.
```

```yaml
apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
 name: example1
 namespace: metallb-system
spec:
 addresses:
 - 192.168.10.0/24
 - 192.168.1.0/24
---
apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
 name: example2
 namespace: metallb-system
spec:
 addresses:
 - 192.155.10.0/24
---
apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
 name: empty

>  namespace: metallb-system

Reviewer 3: Please change the indent from 1 space to 2 spaces, ie,

```yaml
apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
  name: example1
  namespace: metallb-system
spec:
  addresses:
  - 192.168.10.0/24
  - 192.168.1.0/24
---
apiVersion: metallb.io/v1beta1
kind: IPAddressPool
metadata:
  name: example2
  namespace: metallb-system
spec:
  addresses:
  - 192.155.10.0/24
---
apiVersion: metallb.io/v1beta1
kind: L2Advertisement
metadata:
  name: empty
  namespace: metallb-system
```

Author: Done.

---
title: MetalLB load balancer
weight: 3
---

MetalLB is a load balancer designed for bare metal Kubernetes clusters that uses standard routing protocols.

## Prerequisites

- An MKE cluster that does not already have load-balancing functionality.

- A MetalLB-compatible cluster network configuration.

- `kube-proxy` running in `iptables` mode.

- The absence of any cloud provider configuration.

## Configuration

> MetalLB is configured through the `Addons` section of the MKE 4 configuration

Author: ```suggestion
MetalLB is configured through the `addons` section of the MKE 4 configuration
```

file. The function is disabled by default, and thus to use MetalLB you must set
the `enabled` parameter to `true`.

> 1. Obtain the default configuration file:

Author: ```suggestion
1. Obtain the default MKE configuration file:
```
